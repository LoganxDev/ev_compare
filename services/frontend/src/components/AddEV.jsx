import React from 'react';

const AddEV = (props) => {
    return (
        <form onSubmit={(event) => props.addEV(event)}>
            <div className="field">
                <input
                    name="make"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle make"
                    required
                    value={props.make}
                    onChange={props.handleChange}
                />
            </div>
            <div className="field">
                <input
                    name="brand"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle brand"
                    required
                    value={props.brand}
                    onChange={props.handleChange}
                />
            </div>
            <div className="field">
                <input
                    name="base_price"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle base price"
                    required
                    value={props.base_price}
                    onChange={props.handleChange}
                />
            </div>
            <div className="field">
                <input
                    name="storage"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle storage"
                    required
                    value={props.storage}
                    onChange={props.handleChange}
                />
            </div>
            <input
                type="submit"
                className="button is-primary is-large is-fullwidth"
                value="Submit"
            />
        </form>
    )
};

export default AddEV;