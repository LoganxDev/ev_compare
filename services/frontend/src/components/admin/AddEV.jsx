import React from 'react';

// addEV(event) {
//     event.preventDefault();
//     const evData = {
//         make: this.state.make,
//         brand: this.state.brand,
//         base_price: this.state.base_price,
//         storage: this.state.storage
//     }
//     axios.post(`${process.env.REACT_APP_EVS_SERVICE_URL}/evs`, evData)
//     .then((res) => {
//         this.getEvs();
//         this.setState({
//             make: '',
//             brand: '',
//             base_price: 0,
//             storage: 0,
//         });
//         console.log(res)
//     })
//     .catch((err) => {console.log(err)})
// };

const AddEV = (props) => {
    return (
        <form onSubmit={(event) => props.addEV(event)}>
            <div className="field">
                <input
                    name="make"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle make"
                    required
                    value={props.make}
                    onChange={props.handleChange}
                />
            </div>
            <div className="field">
                <input
                    name="brand"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle brand"
                    required
                    value={props.brand}
                    onChange={props.handleChange}
                />
            </div>
            <div className="field">
                <input
                    name="base_price"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle base price"
                    required
                    value={props.base_price}
                    onChange={props.handleChange}
                />
            </div>
            <div className="field">
                <input
                    name="storage"
                    className="input is-large"
                    type="text"
                    placeholder="Enter vehicle storage"
                    required
                    value={props.storage}
                    onChange={props.handleChange}
                />
            </div>
            <input
                type="submit"
                className="button is-primary is-large is-fullwidth"
                value="Submit"
            />
        </form>
    )
};

export default AddEV;