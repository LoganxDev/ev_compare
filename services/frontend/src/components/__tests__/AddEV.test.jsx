import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddEV from '../AddEV';

test('AddEV renders properly', () => {
    const wrapper = shallow(<AddEV />);
    const element = wrapper.find('form');
    expect(element.find('input').length).toBe(5);
    expect(element.find('input').get(0).props.name).toBe('make');
    expect(element.find('input').get(1).props.name).toBe('brand');
    expect(element.find('input').get(2).props.name).toBe('base_price');
    expect(element.find('input').get(3).props.name).toBe('storage');
});

test('AddEV renders a  snapshot properly', () => {
   const tree = renderer.create(<AddEV />).toJSON();
   expect(tree).toMatchSnapshot(); 
});