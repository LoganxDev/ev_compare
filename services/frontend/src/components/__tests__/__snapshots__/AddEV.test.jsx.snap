// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEV renders a  snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="make"
      placeholder="Enter vehicle make"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="brand"
      placeholder="Enter vehicle brand"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="base_price"
      placeholder="Enter vehicle base price"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="storage"
      placeholder="Enter vehicle storage"
      required={true}
      type="text"
    />
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    type="submit"
    value="Submit"
  />
</form>
`;
